src/ussd_public.js

```diff
@@ -474,10 +492,25 @@ go.app = function() {
                         return $(get_error_text(name));
                     }
                 },
-                next: 'state_hiv_messages'
+                next: function(content) {
+                    var year = self.im.user.answers.state_last_period_month.substr(2,4);
+                    var month = self.im.user.answers.state_last_period_month.substr(0,2);
+                    var day = go.utils.double_digit_number(content);
+                    self.im.user.set_answer('last_period_date', year+month+day);
+                    return 'state_get_health_id';
+                }
             });
         });
 
+        self.add('state_get_health_id', function(name) {
+            return go.utils
+                .get_identity(self.im.user.answers.mother_id, self.im)
+                .then(function(identity) {
+                    self.im.user.set_answer('health_id', identity.details.health_id);
```

Comment:
I think we need to assume that something might go wrong with the `health_id` generation here and have at least a check